test/npm/util/stub.js

```diff
@@ -0,0 +1,89 @@
+import {setTimeout} from 'node:timers/promises';
+import test from 'ava';
+import esmock from 'esmock';
+import {_createFixture} from '../../_helpers/stub-execa.js';
+
+/** @type {ReturnType<typeof _createFixture<import('../../../source/npm/util.js')>>} */
+const createFixture = _createFixture('../../../source/npm/util.js', import.meta.url);
+
+test('npm.checkConnection - success', createFixture, [{
+	command: 'npm ping',
+	exitCode: 0,
+}], async ({t, testedModule: npm}) => {
+	t.true(await npm.checkConnection());
+});
+
+test('npm.checkConnection - fail', createFixture, [{
+	command: 'npm ping',
+	exitCode: 1,
+}], async ({t, testedModule: npm}) => {
+	await t.throwsAsync(
+		npm.checkConnection(),
+		{message: 'Connection to npm registry failed'},
+	);
+});
+
+// TODO: find way to timeout without timing out ava
+test.failing('npm.checkConnection - timeout', async t => {
+	const npm = await esmock('../../../source/npm/util.js', {}, {
+		execa: {execa: async () => setTimeout(16_000, {})},
+	});
+
+	await t.throwsAsync(
+		npm.checkConnection(),
+		{message: 'Connection to npm registry timed out'},
+	);
+});
```

Comment:
What's a good way to test that the connection times out? This test times `AVA` out for me locally, but seemingly not on CI.

Reply:
Maybe increase the global AVA timeout?

Reply:
Interestingly, it times out when it’s the only test run, but if the rest of the tests are run at the same time there’s enough time for it to pass.

Reply:
Ended up using `t.timeout`.